graylog2-server/src/main/java/org/graylog/plugins/views/search/db/SearchDbService.java

```diff
@@ -148,18 +117,18 @@ public PaginatedList<Search> findPaginated(DBQuery.Query search, DBSort.SortBuil
      * @param id A Search ID.
      */
     void delete(String id) {
-        db.removeById(new ObjectId(id));
+        mongoUtils.getById(id);
     }
```

Comment:
I think this change is the culprit for #23172 

We're not longer actually removing the DB entries when `delete` is called.